pkgs/native_toolchain_c/test/clinker/throws_test.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:native_assets_cli/native_assets_cli.dart';
+import 'package:native_toolchain_c/native_toolchain_c.dart';
+import 'package:test/test.dart';
+
+import '../helpers.dart';
+
+Future<void> main() async {
+  test(
+    'throws on some platforms',
+    () async {
+      final tempUri = await tempDirForTest();
+
+      final cLinker = CLinker.library(
+        name: 'mylibname',
+        linkerOptions: LinkerOptions.manual(),
+      );
+      await expectLater(
+        () => cLinker.run(
+          config: LinkConfig.build(
+              outputDirectory: tempUri,
+              packageName: 'testpackage',
+              packageRoot: tempUri,
+              targetArchitecture: Architecture.x64,
+              targetOS: OS.linux,
+              buildMode: BuildMode.debug,
+              linkModePreference: LinkModePreference.dynamic,
+              assets: []),
+          output: LinkOutput(),
+          logger: logger,
+        ),
+        throwsUnsupportedError,
+      );
+    },
+    onPlatform: {
```

Comment:
Ah cool! I didn't know about this!

(This structure might not work later if we add support for other platforms, because Windows host/target should not throw, Linux host/target should not throw, but cross Linux-windows host/target should throw. But we can change it then.)